okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/FixedLengthSource.kt

```diff
@@ -1,80 +0,0 @@
-/*
- * Copyright (C) 2024 Square, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package okhttp3.internal
-
-import kotlin.jvm.JvmOverloads
-import okio.Buffer
-import okio.ForwardingSource
-import okio.Source
-
-/**
- * Return a new [Source] whose [read function][Source.read] returns -1 after [byteCount]
- * bytes have been read.
- *
- * @param onReadExhausted Callback invoked once when the end of bytes has been reached. It receives
- * `true` if the end of bytes was because the underlying stream did not contain enough bytes and
- * `false` if [byteCount] bytes were successfully read.
- */
-@JvmOverloads
-internal fun Source.limit(
-  byteCount: Long,
-  onReadExhausted: (eof: Boolean) -> Unit = {},
-): Source {
-  require(byteCount >= 0) { "byteCount < 0: $byteCount" }
-  return FixedLengthSource(this, byteCount, onReadExhausted, truncate = true)
-}
-
-internal class FixedLengthSource(
```

Comment:
This type already existed anyway https://github.com/square/okhttp/blob/master/okhttp%2Fsrc%2FcommonJvmAndroid%2Fkotlin%2Fokhttp3%2Finternal%2Fhttp1%2FHttp1ExchangeCodec.kt and should have been reused not copied.

Reply:
Yup.